src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -612,15 +612,17 @@ public ZonedDateTime getIndividualReleaseDate(Exercise exercise) {
      * @return the time from which on submissions are not allowed, for exercises that are not part of an exam, this is just the due date.
      */
     public ZonedDateTime getIndividualDueDate(Exercise exercise, StudentParticipation participation) {
-        var studentExam = findStudentExam(exercise, participation).orElse(null);
-        // this is the case for all non-exam exercises
-        if (studentExam == null) {
-            return exercise.getDueDate();
+        if (exercise.isExamExercise()) {
+            var studentExam = findStudentExam(exercise, participation).orElse(null);
+            // this is the case for all non-exam exercises
+            if (studentExam == null) {
+                return exercise.getDueDate();
+            }
+            // TODO scale all exercise working times depending on the settings of the exercise, for now, this is just
+            // the individual end date of the exam
```

Comment:
I think we can even remove this. I wrote that back when hadn't decided if we give all exam exercises individual time slots (e.g. first half, second half). Since we are now pretty far away from that, nothing is planned in that regard, and we are happy (AFAIK), we can remove the comment.
```suggestion
```

It is probably only confusing for newer developers.